lago/virt.py

```diff
@@ -114,6 +115,51 @@ def _create_vm(self, vm_spec):
         vm_spec['vm-type'] = vm_type_name
         return vm_type(self, vm_spec)
 
+    def check_memory_requirements(self, conf):
```

Comment:
Thanks for the review, David.

> this is not so simple, now we have vm-providers,

Right, noticed your change introducing that functionality.

> meaning that not all the vms might be on the local libvirt, so the required amount of memory is not the total sum of all of them (some might not even declare a memory entry, like the ones using the ssh provider)

Good catch. Will inspect the SSH specific case to see what can be done, when Lago config doesn't contain memory entries and if it could be handled somehow.

If impossible to implement for all the providers, will close this PR as irrelevant.